### PR TITLE
Create new application via API

### PR DESCRIPTION
**Pull request by LAACharlesBoardman**
Added webmock to test API

### CODE REVIEW

File: spec/services/application_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      ApplicationService.create
+
+      assert_requested :post, ENV['API_APPLICATION_URL'] + '/v1/applications',
+        times: 1
+    end
+
+    it 'returns the application ref' do
+
+      expect(ApplicationService.create).to eq(application_ref)
+    end
```
Comment (adrianclay) [inline spec/services/application_service_spec.rb]: Be good to define some behaviour in the event that the API is unavailable.  Does the team know how we'd like the frontends to behave if the API is down?

File: app/services/application_service.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'json'
+
+class ApplicationService
+
+  def self.create
+    uri = URI.parse(ENV['API_APPLICATION_URL'] + '/v1/applications')
+    req = Net::HTTP::Post.new(uri.path)
+
+    res = Net::HTTP.start(uri.host, uri.port) do |http|
```
Comment (adrianclay) [inline app/services/application_service.rb]: ```Net::HTTP``` is a basic approach to making HTTP requests, but works for now.

I know @DamanFergus has some HTTP libraries that he could share.

**Comment by adrianclay** [comment]
It's been really great doing mob programming today on this.  Hope you've all learnt something about testing API calls, and got a more realistic view of what TDD can look like for production code.

**Comment by adrianclay** [comment]
We'll need a ticket to deal with Environment variables, is this something that is already included in another ticket or do we need to create a new one CC @mpw5 ?

File: spec/services/application_service_spec.rb (hunk 1 above)
Comment (adrianclay) [reply spec/services/application_service_spec.rb]: Similarly what behaviour do we want when the API is giving us errors, 40X, 50X...

Comment (bleach) [reply spec/services/application_service_spec.rb]: I think, for now, all HTTP errors from the API should cause the frontend to fail and return HTTP 500.

Comment (a5rar) [reply spec/services/application_service_spec.rb]: We may down the road want to create a static page for the 500, probably one for Simon to look at.
As for 40X thats usually bit more interesting.
For validation errors its good to just relay these onto the front end.

We may encounter a 401 not authenticated or a 403 not authorised.  I think we should discuss these at some point what behavior we want  the front end to have in these scenarios

**Pull request merged**